src/App.tsx

```diff
@@ -5,21 +5,37 @@ import AddNewPlant from "./pages/addNewPlantPage/AddNewPlant.tsx";
 import GlobalStyle from "./theme/globalStyles.ts";
 import { theme } from "./theme/theme.ts";
 import { ThemeProvider } from "styled-components";
+import { FormValues } from "./components/form/Form.types.ts";
+
+const App = () => {
+  const token = localStorage.getItem("formData");
+
+  const plantsCollection: FormValues[] =
+    (token && JSON.parse(localStorage.getItem("formData") || "")) || [];
 
-function App() {
   return (
     <>
       <ThemeProvider theme={theme}>
         <GlobalStyle />
         <Routes>
           <Route path={"/"} element={<MyCollectionPage />} />
           <Route path={"/dodaj-roślinkę"} element={<AddNewPlant />} />
-          <Route path={"/monstera/*"} element={<PlantDetailPage />} />
+
+          {plantsCollection.map((item) => {
+            return (
+              <Route
+                key={item.plantID}
+                path={`/${item.plantName}/*`}
+                element={<PlantDetailPage currentPlantId={item.plantID} />}
+              />
+            );
+          })}
+
```

Comment:
this iteration isn't perfect, but can stay until you implement the external API. 
Does it work for you though? Clicking on the plant I've just created does not take me to its details page. Where's the navigation?